File: src/models/moment.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug)]
pub enum Moment {
  Morning,
  Lunch,
  Evening,
}

impl Moment {
  pub fn from_api(str: &str) -> Moment {
    match str {
      "matin" => Moment::Morning,
      "midi" => Moment::Lunch,
      "soir" => Moment::Evening,
      _ => unreachable!("Unknown opening: {}", str),
    }
  }
}

File: library/src/commonMain/kotlin/decoders/moment.kt
package decoders

import models.Moment

fun decodeMoment (value: String): Moment {
  return when (value) {
    "matin" -> Moment.MORNING
    "midi" -> Moment.LUNCH
    "soir" -> Moment.EVENING
    else -> throw Exception("Unknown moment: $value")
  }
}
